Add XML documentation to eliminate all CS1591 warnings

Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(20,14): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(22,19): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.Url' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(23,19): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.Username' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(24,19): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.Password' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(26,12): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.GenericConnection(string, string, string)' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Helpers/MacOSHelper.cs(31,24): warning CS1591: Missing XML comment for publicly visible type or member 'MacOSHelper.IsMacOS' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(56,21): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.Network' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(103,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithMainRemoteArtifacts(params string[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(109,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithMainRemoteArtifacts(params RemoteArtifact[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(115,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSecondaryArtifacts(params string[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(121,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSecondaryRemoteArtifacts(params RemoteArtifact[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(127,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSnapshots(params string[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(133,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSecrets(params Secret[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
  Microcks.Testcontainers -> /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/bin/Release/net8.0/Microcks.Testcontainers.dll
  Microcks.Testcontainers.Tests -> /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/tests/Microcks.Testcontainers.Tests/bin/Release/net8.0/Microcks.Testcontainers.Tests.dll

Build succeeded.

Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(20,14): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(22,19): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.Url' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(23,19): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.Username' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(24,19): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.Password' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Connection/GenericConnection.cs(26,12): warning CS1591: Missing XML comment for publicly visible type or member 'GenericConnection.GenericConnection(string, string, string)' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Helpers/MacOSHelper.cs(31,24): warning CS1591: Missing XML comment for publicly visible type or member 'MacOSHelper.IsMacOS' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(56,21): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.Network' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(103,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithMainRemoteArtifacts(params string[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(109,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithMainRemoteArtifacts(params RemoteArtifact[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(115,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSecondaryArtifacts(params string[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(121,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSecondaryRemoteArtifacts(params RemoteArtifact[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(127,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSnapshots(params string[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
Warning: /home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs(133,38): warning CS1591: Missing XML comment for publicly visible type or member 'MicrocksContainerEnsemble.WithSecrets(params Secret[])' [/home/runner/work/microcks-testcontainers-dotnet/microcks-testcontainers-dotnet/src/Microcks.Testcontainers/Microcks.Testcontainers.csproj::TargetFramework=net8.0]
    13 Warning(s)
    0 Error(s)


currently there are 13 warnings, this issue will track in fixing them.

--- a/src/Microcks.Testcontainers/Connection/GenericConnection.cs
+++ b/src/Microcks.Testcontainers/Connection/GenericConnection.cs
@@ -17,12 +17,32 @@
 
 namespace Microcks.Testcontainers.Connection;
 
+/// <summary>
+/// Represents generic connection information (URL and optional credentials).
+/// </summary>
 public class GenericConnection
 {
+    /// <summary>
+    /// Gets the connection URL.
+    /// </summary>
     public string Url { get; }
+
+    /// <summary>
+    /// Gets the username used for authentication, if any.
+    /// </summary>
     public string Username { get; }
+
+    /// <summary>
+    /// Gets the password used for authentication, if any.
+    /// </summary>
     public string Password { get; }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GenericConnection"/> class.
+    /// </summary>
+    /// <param name="url">The connection URL.</param>
+    /// <param name="username">The username used for authentication, if any.</param>
+    /// <param name="password">The password used for authentication, if any.</param>
     public GenericConnection(string url, string username = null, string password = null)
     {
         Url = url;

--- a/src/Microcks.Testcontainers/Helpers/MacOSHelper.cs
+++ b/src/Microcks.Testcontainers/Helpers/MacOSHelper.cs
@@ -28,6 +28,9 @@ namespace Microcks.Testcontainers.Helpers;
 /// </summary>
 public static class MacOSHelper
 {
+    /// <summary>
+    /// Gets a value indicating whether the current process is running on macOS (Darwin) on Arm64.
+    /// </summary>
     public static bool IsMacOS { get; internal set; }
         = RuntimeInformation.ProcessArchitecture == Architecture.Arm64
           && RuntimeInformation.OSDescription.Contains("Darwin");

--- a/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs
+++ b/src/Microcks.Testcontainers/MicrocksContainerEnsemble.cs
@@ -51,6 +51,9 @@ public class MicrocksContainerEnsemble : IAsyncDisposable, IArtifactAndSnapshotM
 
     private readonly INetwork _network;
 
+    /// <summary>
+    /// Gets the Docker network used by this ensemble.
+    /// </summary>
     public INetwork Network { get => this._network; }
 
 
@@ -98,36 +101,66 @@ public class MicrocksContainerEnsemble : IAsyncDisposable, IArtifactAndSnapshotM
         return this;
     }
 
+    /// <summary>
+    /// Configures the Microcks container ensemble with the specified main remote artifacts.
+    /// </summary>
+    /// <param name="mainRemoteArtifacts">An array of remote artifact URIs/locations to be used by the Microcks container.</param>
+    /// <returns>The updated <see cref="MicrocksContainerEnsemble"/> instance.</returns>
     public MicrocksContainerEnsemble WithMainRemoteArtifacts(params string[] mainRemoteArtifacts)
     {
         this._microcksBuilder.WithMainRemoteArtifacts(mainRemoteArtifacts);
         return this;
     }
 
+    /// <summary>
+    /// Configures the Microcks container ensemble with the specified main remote artifacts.
+    /// </summary>
+    /// <param name="remoteArtifacts">The remote artifact definitions to be used by the Microcks container.</param>
+    /// <returns>The updated <see cref="MicrocksContainerEnsemble"/> instance.</returns>
     public MicrocksContainerEnsemble WithMainRemoteArtifacts(params RemoteArtifact[] remoteArtifacts)
     {
         this._microcksBuilder.WithMainRemoteArtifacts(remoteArtifacts);
         return this;
     }
 
+    /// <summary>
+    /// Configures the Microcks container ensemble with the specified secondary artifacts.
+    /// </summary>
+    /// <param name="secondaryArtifacts">An array of secondary artifact file names to be used by the Microcks container.</param>
+    /// <returns>The updated <see cref="MicrocksContainerEnsemble"/> instance.</returns>
     public MicrocksContainerEnsemble WithSecondaryArtifacts(params string[] secondaryArtifacts)
     {
         this._microcksBuilder.WithSecondaryArtifacts(secondaryArtifacts);
         return this;
     }
 
+    /// <summary>
+    /// Configures the Microcks container ensemble with the specified secondary remote artifacts.
+    /// </summary>
+    /// <param name="remoteArtifacts">The remote artifact definitions to be used by the Microcks container.</param>
+    /// <returns>The updated <see cref="MicrocksContainerEnsemble"/> instance.</returns>
     public MicrocksContainerEnsemble WithSecondaryRemoteArtifacts(params RemoteArtifact[] remoteArtifacts)
     {
         this._microcksBuilder.WithSecondaryRemoteArtifacts(remoteArtifacts);
         return this;
     }
 
+    /// <summary>
+    /// Configures the Microcks container ensemble with the specified snapshots.
+    /// </summary>
+    /// <param name="snapshots">An array of snapshot file names to be loaded into the Microcks container.</param>
+    /// <returns>The updated <see cref="MicrocksContainerEnsemble"/> instance.</returns>
     public MicrocksContainerEnsemble WithSnapshots(params string[] snapshots)
     {
         this._microcksBuilder.WithSnapshots(snapshots);
         return this;
     }
 
+    /// <summary>
+    /// Configures the Microcks container ensemble with the specified secrets.
+    /// </summary>
+    /// <param name="secrets">The secrets to create in the Microcks container.</param>
+    /// <returns>The updated <see cref="MicrocksContainerEnsemble"/> instance.</returns>
     public MicrocksContainerEnsemble WithSecrets(params Model.Secret[] secrets)
     {
         this._microcksBuilder.WithSecrets(secrets);

